src/Auth/UnattendedAuthentication.cs

```diff
@@ -0,0 +1,61 @@
+﻿using IdentityModel.Client;
+using System;
+using System.Net.Http;
+using System.Threading.Tasks;
+
+namespace Jpp.Common.Backend.Auth
+{
+    public class UnattendedAuthentication : BaseOAuthAuthentication
+    {
+        protected string _clientId, _clientSecret, _username, _password;
```

Comment:
Are these not private?

Reply:
Protected to allow for a derived class to access/modify as required

Reply:
If that's the case then properties are probably required to ensure say if username is change after authentication is set, then the user is not still deemed authenticated.